archive/miu_language/basic.lean

```diff
@@ -0,0 +1,215 @@
+/-
+Copyright (c) 2020 Gihan Marasingha. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Gihan Marasingha
+-/
+import tactic.linarith
+
+/-!
+# An MIU Decision Procedure in Lean
+
+The [MIU formal system](https://en.wikipedia.org/wiki/MU_puzzle) was introduced by Douglas
+Hofstadter in the first chapter of his 1979 book,
+[Gödel, Escher, Bach](https://en.wikipedia.org/wiki/G%C3%B6del,_Escher,_Bach).
```

Comment:
Note that we have a `.bib` file in the repo. If you want you can add an entry there as well.

Reply:
OK! I note there isn't an entry for TPIL. Is there a reason for that? It might be nice to refer to TPIL or another resource on inductive types.

Reply:
It's certainly fine to add things to the bibliography! We should use it much more than we have been.

Reply:
If I'm to edit `references.bib`, should I create a new PR or add the commit to this PR?

In the latter case, is it OK to `git merge master` into the miu branch or should I rebase (or something else)? Thanks!

Reply:
I'd say you can just add the commit to this PR and it's better to merge master into this branch. Rebasing is generally discouraged after people start reviewing since it can screw up GitHub's review comments.

Reply:
Thanks. I'm still finding my way around Git! I'm about to push the merged branch with my additions to `references.bib` and the corresponding addition of citations in `basic.lean`.